src/ol/style/expressions.js

```diff
@@ -590,6 +591,29 @@ Operators['!'] = {
     return `(!${expressionToGlsl(context, args[0])})`;
   },
 };
+
+function getDecisionOperator(operator) {
+  return {
+    getReturnType: function (args) {
+      return ValueTypes.BOOLEAN;
+    },
+    toGlsl: function (context, args) {
+      assertArgsMinCount(args, 2);
+      for (let i = 0; i < args.length; i++) {
+        assertBoolean(args[i]);
+      }
+      let result = '';
+      result = args
+        .map((arg) => expressionToGlsl(context, arg))
+        .join(` ${operator} `);
+      result = `(${result})`;
+      return result;
+    },
+  };
+}
+
+Operators['and'] = getDecisionOperator('&&');
```

Comment:
Looks like we're back with the name `and` instead of `all`... did something go wrong when squashing the commits?